metricflow/protocols/sql_client.py

```diff
@@ -24,6 +26,26 @@ class SqlEngine(Enum):
     DATABRICKS = "Databricks"
     TRINO = "Trino"
 
+    @property
+    def unsupported_granularities(self) -> Set[TimeGranularity]:
+        """Granularities that can't be used with this SqlEngine."""
```

Comment:
Per discussion, all looks good with this list of limitations. Let's just update the documentation to explain why these are in place. Maybe something like:
```suggestion
        """Granularities that can't be used with this SqlEngine.

        We allow the smallest granularity the SQLEngine supports for its base TIMESTAMP type and all our required
        operations (e.g., DATE_TRUNC). For example, when we added support for these granularities
        Trino supported more precise types for storage and access, but Trino's base TIMESTAMP type and
        date_trunc function only supported millisecond precision.
        """
```